flink-connectors/flink-connector-pubsub/pom.xml

```diff
@@ -0,0 +1,90 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.apache.flink</groupId>
+		<artifactId>flink-connectors</artifactId>
+		<version>1.8-SNAPSHOT</version>
+		<relativePath>..</relativePath>
+	</parent>
+
+	<artifactId>flink-connector-pubsub_${scala.binary.version}</artifactId>
+	<name>flink-connector-pubsub</name>
+
+	<packaging>jar</packaging>
+
+	<dependencyManagement>
+		<dependencies>
+			<dependency>
+				<!-- This is the way we get a consistent set of versions of the Google tools -->
+				<groupId>com.google.cloud</groupId>
+				<artifactId>google-cloud-bom</artifactId>
+				<version>0.70.0-alpha</version>
+				<type>pom</type>
+				<scope>import</scope>
+			</dependency>
+		</dependencies>
+	</dependencyManagement>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-streaming-java_${scala.binary.version}</artifactId>
+			<version>${project.version}</version>
+			<scope>provided</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>com.google.cloud</groupId>
+			<artifactId>google-cloud-pubsub</artifactId>
+			<!-- Version is pulled from google-cloud-bom -->
+			<exclusions>
+				<!-- Exclude an old version of guava that is being pulled
+                in by a transitive dependency of google-api-client -->
+				<exclusion>
+					<groupId>com.google.guava</groupId>
+					<artifactId>guava-jdk5</artifactId>
+				</exclusion>
+			</exclusions>
+			<scope>provided</scope>
```

Comment:
Why is the scope of this dependency set to "provided"?
I'm currently trying out the pull request and my IntelliJ is not able to find the pubsub dependencies (such as the `Credentials` from Google)

Reply:
Great thanks for having a look at this again :) This weekend I was looking to see if I could upgrade this branch to 1.8. I'll try to push those changes today or tomorrow.

I've set this scope to provided because the google sdk's give a lot of dependency problems. So in our case we would always provide our own correct version (using bom for example https://github.com/googleapis/google-cloud-java/tree/master/google-cloud-bom)

Would you say it's better to just include it and if anyone has dependency problems they should just exclude the dependency?

Reply:
Does setting the dependencies to `compile` scope lead to conflicts with Flink's dependencies or with your own project?

I'm always in favor of a good "out of the box" experience, so that somebody who's just following the documentation page is able to get the examples running without tweaking dependencies manually.
Properly integrating the dependency with your own project dependencies will always be involved for such a connector.

So if it doesn't conflict with Flink, I would remove the `provided` scope.

Reply:
Looking forward to the code push :)

Reply:
Done, I rebased the our commits on upstream master

Reply:
Done